frontend/app/reducers/catReducer.js

```diff
@@ -0,0 +1,29 @@
+import { REQUEST, SUCCESS, FAILURE } from '../actions/catActions'
+
+const initialState= {
+  isFetching: false,
+  randomCat: {}
+}
+
+function randomCat(state = initialState, action) {
+  switch (action.type) {
+    case REQUEST:
+      return Object.assign({}, state, {
+        isFetching: true
+    })
+    case FAILURE:
+      return Object.assign({}, state, {
+        isFetching: false,
+        error: action.payload
+    })
```

Comment:
Right now just using default behavior of the Failure FSA. If you change the endpoint string in the action to something else that will 404, the state will be updated with the 404 error message. I will look more into error handling in another ticket.